app/dashboard/notifications.py

```diff
@@ -394,196 +393,147 @@ def get_status_header(bounty):
 
 
-def maybe_market_to_github(bounty, event_name, profile_pairs=None):
+@build_github_notification
+def maybe_market_to_github(bounty, msg, event_name):
     """Post a Github comment for the specified Bounty.
 
     Args:
         bounty (dashboard.models.Bounty): The Bounty to be marketed.
         event_name (str): The name of the event.
-        profile_pairs (list of tuples): The list of username and profile page
-            URL tuple pairs.
 
     Returns:
         bool: Whether or not the Github comment was posted successfully.
 
     """
-    if not bounty.is_notification_eligible(var_to_check=settings.GITHUB_CLIENT_ID):
+    if msg is False or not bounty.is_notification_eligible(var_to_check=settings.GITHUB_CLIENT_ID):
         return False
 
     # Define posting specific variables.
-    comment_id = None
     url = bounty.github_url
     uri = parse(url).path
     uri_array = uri.split('/')
 
     # Prepare the comment message string.
-    msg = build_github_notification(bounty, event_name, profile_pairs)
-    if not msg:
-        return False
+    if msg == bounty.comment_id:
+        delete_issue_comment(msg, username, repo)
 
+    # Try either posting or patching the comment on the github issue
     try:
         username = uri_array[1]
         repo = uri_array[2]
         issue_num = uri_array[4]
 
-        if event_name == 'work_started':
-            comment_id = bounty.interested_comment
-        elif event_name in ['work_done', 'work_submitted']:
-            comment_id = bounty.submissions_comment
-
-        # Handle creating or updating comments if profiles are provided.
-        if event_name in ['work_started', 'work_submitted'] and profile_pairs:
-            if comment_id is not None:
-                patch_issue_comment(comment_id, username, repo, msg)
-            else:
-                response = post_issue_comment(username, repo, issue_num, msg)
-                if response.get('id'):
-                    if event_name == 'work_started':
-                        bounty.interested_comment = int(response.get('id'))
-                    elif event_name in ['work_done', 'work_submitted']:
-                        bounty.submissions_comment = int(response.get('id'))
-                    bounty.save()
-        # Handle deleting comments if no profiles are provided.
-        elif event_name in ['work_started'] and not profile_pairs:
-            if comment_id:
-                delete_issue_comment(comment_id, username, repo)
-                if event_name == 'work_started':
-                    bounty.interested_comment = None
-                elif event_name == 'work_done':
-                    bounty.submissions_comment = None
-                bounty.save()
-        # If this isn't work_started/done, simply post the issue comment.
+        print(bounty.comment_id)
+        if bounty.comment_id:
+            response = patch_issue_comment(bounty.comment_id, username, repo, msg)
```

Comment:
i worry that if we only patch issues then no new github notifications will go out when theres an update..

Reply:
@PixelantDesign @mbeacom Open to your feedback here!